website/docs/reference/appsmith-framework/widget-actions/post-message.md

```diff
@@ -0,0 +1,77 @@
+# Post Message
+
+The **Post message** action is used to enable safe cross-origin communication between [Window](https://developer.mozilla.org/en-US/docs/Web/API/Window) objects; for example, between a page and a pop-up that it has created, or between a page and an iframe embedded within it. The Appsmith function `postWindowMessage` works by using the JavaScript function [`postMessage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) so that you can communicate between applications in situations such as:
+
+- Appsmith being embedded in an iframe of a parent application.
+- Appsmith using [Iframe](/reference/widgets/iframe/) widgets to embed other applications or pages.
+
+## Signature
+
+```javascript
+postWindowMessage(message, targetIframe, targetOrigin)
+```
+
+### Arguments
+
+| Argument Name | Description |
+| ------------- | ----------- |
+| **message** | This is the message to send to the parent application. Most JavaScript values are acceptable here, except `null` and `undefined`. (Default: `""`) |
+| **targetIframe** | This is the window to which you want to send the message. If its value is `"window"`, you are sending a message to the parent application’s window (where Appsmith is embedded). If you want to send a message to an iframe within Appsmith, enter the name of the iframe here. (Default: `"window"`) |
+| **targetOrigin** | This is the URL to which you can send messages. The default value of "`*`" means the message can be sent to any URL. If you want to limit sending messages to only the parent application (in which Appsmith is embedded), enter the URL of the parent application here. (Default: `"*"`) |
+
+## Examples
+
+### Page embedded in Appsmith
+
+For these examples, imagine you are building an app in Appsmith that has an Iframe widget called `Iframe1` containing an embedded page.
```

Comment:
delete this line.  Here we can explain the scenario a bit deeper  - a page or app embedded in an Appsmith iframe widget.......